api/transport/inboundtlsmode.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2022 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package transport
+
+import (
+	"encoding"
+	"fmt"
+)
+
+const (
+	// Disabled TLS mode accepts plaintext connections only.
+	Disabled InboundTLSMode = iota
+
+	// Permissive TLS mode inbound accepts both TLS and plaintext connections.
+	Permissive
+
+	// Enforced TLS mode inbound accepts TLS connections only.
+	Enforced
+)
+
+var (
+	_ fmt.Stringer             = (*InboundTLSMode)(nil)
+	_ encoding.TextUnmarshaler = (*InboundTLSMode)(nil)
+)
+
+// InboundTLSMode represents the TLS mode of the inbound.
+type InboundTLSMode uint16
+
+// String returns a lower-case ASCII representation of TLS mode.
+func (t InboundTLSMode) String() string {
```

Comment:
sounds this should be part of stringer generation: https://pkg.go.dev/golang.org/x/tools/cmd/stringer